superset/datasets/dao.py

```diff
@@ -287,24 +296,27 @@ def find_dataset_metric(
         return db.session.query(SqlMetric).get(metric_id)
 
     @classmethod
-    def delete_metric(
-        cls, model: SqlMetric, commit: bool = True
-    ) -> Optional[TableColumn]:
```

Comment:
Should have been `SqlMetric` and not `TableColumn`.

Reply:
Why does the command still return the metric when it is deleted?

Reply:
yeah, prob not needed